### DISCUSSION TITLE
Organization#base/identifier.system - Идентификаторы организации

### DISCUSSION THREAD

**Discussion by niquola** [fhir-ru/core#41] [Административная РГ]
Имеет смысл сделать extensible ValueSet - `fhir:ru:org-id` для `Organization.identifier.system`
Возможно вместе с одноименным CodeSystem

Первый набросок:

System: `fhir:ru:org-id`

| Code      | Description | OID |
| ----------- | ----------- |----|
| `fhir:ru:frmo:id`     | УИ организации по ФРМО        |  1.2.643.5.1.13.13.12.2 | 
| `fhir:ru:org-inn`      | ИНН юридического лица        |  1.2.643.3.131.1.1 | 
| `fhir:ru:ogrn`      | ОГРН (основной государственный регистрационный номер)       | 1.2.643.100.1  |

Что еще упустил?

## Вопросы:

нужно соглашение по system - URL или URI:

Мне больше нравится URI - не надо протокол добавлять http https?
Для URI - нужно договорится про префикс - `fhir:ru`,   `ru:fhir`, just `ru`

**Comment by toropok** [comment]
А в чем смысл этого набора значений?

Непонятно что вы пытаетесь добиться его созданием, а главное почему это как то относится к ru-core? и как это связано с идентификатором организации, учитывая что у каждого из этих справочников (ФРМО, ФНС) есть уже свои OID'ы

**Reply by niquola** [reply]
смысл договорится про возможный набор значений для этого поля, чтобы в будущем все знали какой system проставлять для определенного типа идентификатора

Как положить ИНН? Как положить ОГРН?

**Reply by niquola** [reply]
@toropok  Про OIDы существующих справочников - можешь сcылочу или выдержку сюда добавить?

**Reply by niquola** [reply]
Смысл rucore - договорится о том как вещи которые можно сделать несколькими способами делать одним общим

Предлагай другой вариант.

Проверочный вопросы:

* Как в организацию положить ОГРН?
* Как добавить новый тип идентификаторов организации ?

**Reply by ai-fhir** [reply]
Про OID стандарт нам говорит [следующее](http://www.hl7.org/fhir/datatypes.html#Identifier) "Resolvable URLs are generally preferred by implementers over non-resolvable URNs, particularly opaque URNs such as OIDs (urn:oid:) or UUIDs (urn:uuid:)." 

Для US Core были введены новые URI для ssn в `Identifier.system`, при этом в этом OID для данного значения уже [был](http://www.hl7.org/fhir/identifier-registry.html#4.6).

**Comment by toropok** [comment]
это гуглится просто типа "OID инн" 1.2.643.3.131.1.1 = ИНН, 1.2.643.100.1 = ОГРН, так как эти справочники и задачи решались еще лет 10-15 назад для ЭЦП

смысле вводить какой то спец вальюсет не вижу. эти OID будут важны для профилилования ресурсов под ru-core, а просто его хранить так и непонятно для чего.

**Reply by niquola** [reply]
valueset все равно надо завести - чтобы не гуглить и машины могли валидировать
OID конечно уродливые ребята - я бы еще поголосовал за семантические code

**Reply by niquola** [reply]
valueset и делается для профилирования - ты потом его байндишь к элементу

**Reply by Vrodvi** [reply]
Справочники и системы идентификации объектов - не одно и то же.
Реестры с идентификаторами объектов (Identifier Systems) могут вестись отдельно, не в виде CodeSystem\ValueSet, что уже происходит в случае с ИНН и ОГРН.
То что реестр мед.организаций (ФРМО) продублирован в виде справочника на НСИ - это скорее исключение.

При профилировании байндинг valueset для идентификаторов не делается  - давайте осторожнее использовать терминологию стандарта FHIR, вот пример определения системы идентификации из FHIR US Core, это не байндинг на valuset:
![image](https://user-images.githubusercontent.com/92928137/143551489-6814d693-8a26-4ea1-aed5-6d5ee14e4708.png)

Указанный uri `http://hl7.org/fhir/sid/us-npi` можно найти как раз на странице с описанием систем идентификации:
http://www.hl7.org/implement/standards/fhir/identifier-registry.html
, а не в виде значения из valueset

**Reply by niquola** [reply]
Да в us-core не binding - у них один слайс по system 
В нашем случае мне кажется valueset на system хорошая идея

**Reply by niquola** [reply]
Есть еще http://www.hl7.org/fhir/namingsystem.html

> The CodeSystem resource defines the content of a code system, and also its preferred identifier. The NamingSystem resource identifies the existence of a code or identifier system, and its possible and preferred identifiers. The key difference between the resources is who creates and manages them - CodeSystem resources are managed by the owner of the code system resource, who can properly define the features and content of the code system. NamingSystem resources, on the other hand, are frequently defined by 3rd parties that encounter the code system in use, and need to describe the use, but do not have the authority to define the features and content. Additionally, there may be multiple authoritative NamingSystem resources for a code system, but there should only be one CodeSystem resource.

**Reply by iradio** [reply]
Согласен в необходимости дать образец кодирования `Organization.identifier.system`, но представляется очевидным, что в этом поле могут быть не только справочные значения, но и локальные расширения. Однако об общеупотребимых надо договориться и вынести в федеральное НСИ.
Был такой справочник в федеральном НСИ https://nsi.rosminzdrav.ru/#!/refbook/1.2.643.5.1.13.2.1.1.685/version/1.0 , именно оттуда в указанные выше OIDы ИНН и других.
![image](https://user-images.githubusercontent.com/1135903/143592431-105dbbc8-39bd-4cdd-95f7-d6b343087a11.png)
Подождем прихода сюда коллег из регламентной службы ЦНИИОИЗ с ответом на вопрос - есть ли актуальная замена справочнику.

**Comment by Vrodvi** [comment]
Напрашивается создание сайта со спецификацией российского корневого руководства по реализации, где должен быть заведён раздел с перечнем систем идентификации, как это сделано в самом стандарте FHIR:
- [FHIR identifier-registry](http://www.hl7.org/implement/standards/fhir/identifier-registry.html#)

<html>
<body>


identifier | URI | OID (for non-FHIR systems) | Type | Comment
-- | -- | -- | -- | --
| Глобально уникальный идентификатор медицинской организации по Реестру медицинских организаций Российской Федерации| urn:oid:1.2.643.5.1.13.13.11.1461 | 1.2.643.5.1.13.13.11.1461 |   |   |
| Идентификационный номер налогоплательщика (ИНН ) | urn:oid:1.2.643.3.131.1.1 | 1.2.643.3.131.1.1 |   |   |
| Основной государственный регистрационный номер (ОГРН) | urn:oid:1.2.643.100.1 | 1.2.643.100.1 |   |   |


</body>
</html>

**Reply by niquola** [reply]
Ага звучит неплохо

**Reply by niquola** [reply]
Только это просто страничка - те у нее нет machine readable представления - CodeSystem возможно подойдет

**Reply by Vrodvi** [reply]
Можно дублировать эту таблицу как CodeSystem, это добавит удобства в обработке, но в руководстве по реализации (FHIR IG Ru core) это точно должно быть вынесено в отдельный раздел - "системы идентификации".

**Comment by iradio** [comment]
На текущий момент ЕГИСЗ предлагает использовать вот такой справочник для кодового обозначения системы. Назначение справочника может быть расширено.
https://nsi.rosminzdrav.ru/#!/refbook/1.2.643.5.1.13.13.99.2.311
![image](https://user-images.githubusercontent.com/1135903/143609493-e6c2a007-6ded-4bf8-8bf2-e0ba899e32ea.png)

**Reply by niquola** [reply]
А для организаций есть такой?
